source/extensions/filters/network/http_connection_manager/config.cc

```diff
@@ -54,6 +57,53 @@ FilterFactoryMap::const_iterator findUpgradeCaseInsensitive(const FilterFactoryM
   return upgrade_map.end();
 }
 
+std::unordered_map<std::string, std::string>
+convertJsonFormatToMap(ProtobufWkt::Struct json_format) {
+  std::unordered_map<std::string, std::string> output;
+  for (const auto& pair : json_format.fields()) {
+    if (pair.second.kind_case() != ProtobufWkt::Value::kStringValue) {
+      throw EnvoyException("Only string values are supported in the JSON access log format.");
+    }
+    output.emplace(pair.first, pair.second.string_value());
+  }
+  return output;
+}
+
+AccessLog::FormatterPtr createFormatter(const Protobuf::Message& config,
```

Comment:
This might be a good fuzz target; CC @asraa

Reply:
Do you reckon we could add a fuzzer in the context of this review? There are quite a few examples now in Envoy, so if you can provide this it'd be great.

Reply:
Here I'm reusing existing `AccessLog:Formatter` which has already fuzzer tests. Maybe should I add more test cases to existing one?

Reply:
That's true actually, I think I may have even wrote it :) Thank for the reminder.